validator/src/journal/block_validator.rs

```diff
@@ -70,3 +72,118 @@ impl<'source> FromPyObject<'source> for BlockValidationResult {
         })
     }
 }
+
+trait StateBlockValidation {
```

Comment:
Could you add a doc comment to these traits, so that I can differentiate between the their behaviour and output?

Reply:
I can add a doc comment to the traits. Their main behavior is that they return `ValidationError::BlockValidationFailure` if the block fails that check. Something that I thought of is that there could be a single `BlockValidation` trait with generic `Result<R, ValidationError>` return value. What do you think of that?

Reply:
The generic return type turned out to be a bad idea.